pkg/api/network/store.go

```diff
@@ -0,0 +1,78 @@
+package network
+
+import (
+	"encoding/json"
+	"errors"
+	"strconv"
+
+	cniv1 "github.com/containernetworking/cni/pkg/types"
+	"github.com/rancher/apiserver/pkg/apierror"
+	"github.com/rancher/apiserver/pkg/types"
+	cni "github.com/rancher/harvester/pkg/generated/controllers/k8s.cni.cncf.io/v1"
+	"github.com/rancher/wrangler/pkg/schemas/validation"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+type networkStore struct {
+	types.Store
+	nadCache cni.NetworkAttachmentDefinitionCache
+}
+
+type NetConf struct {
+	cniv1.NetConf
+	BrName       string `json:"bridge"`
+	IsGW         bool   `json:"isGateway"`
+	IsDefaultGW  bool   `json:"isDefaultGateway"`
+	ForceAddress bool   `json:"forceAddress"`
+	IPMasq       bool   `json:"ipMasq"`
+	MTU          int    `json:"mtu"`
+	HairpinMode  bool   `json:"hairpinMode"`
+	PromiscMode  bool   `json:"promiscMode"`
+	Vlan         int    `json:"vlan"`
+}
+
+const (
+	FieldSpec   = "spec"
+	FieldConfig = "config"
+)
+
+func (s *networkStore) Create(request *types.APIRequest, schema *types.APISchema, data types.APIObject) (types.APIObject, error) {
+	objData := data.Data()
+	config := objData.Map(FieldSpec).String(FieldConfig)
+	if config == "" {
+		return types.APIObject{}, apierror.NewAPIError(validation.InvalidBodyContent, "NAD config %s is empty")
+	}
+
+	var bridgeConf = &NetConf{}
+	err := json.Unmarshal([]byte(config), &bridgeConf)
+	if err != nil {
+		return types.APIObject{}, apierror.NewAPIError(validation.ServerError, "Failed to decode NAD config value, error: "+err.Error())
+	}
+
+	isUnique, err := s.checkUniqueVlanID(data.Namespace(), bridgeConf)
+	if !isUnique {
```

Comment:
Check the err. It includes both API call error and duplicate vlan error. isUnique may be unnecessary , nil error indicates the valid case.